flake.nix

```diff
@@ -15,13 +15,20 @@
   };
 
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.05";
+    #nixpkgs.url = "github:nixos/nixpkgs/nixos-23.05";
+    #BUG: https://github.com/NixOS/nixpkgs/issues/235179
+    #BUG: https://github.com/NixOS/nixpkgs/issues/235526
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-22.11";
```

Comment:
In particular, we can't merge this as it will break `nixpkgs` via `inputs` for all other targets by taking us back to `22.11` when we have just put the effort to move to `23.05`.

Reply:
One possible solution is to introduce another input - like `nixpkgs-2211` and pass that to icicle targets instead of `nixpkgs`. The same with `nixos-hardware` below where your fork would be additional input - at least until your upstream PR is merged.
There are some downside from that, namely duplicating the inputs on some areas, but that proposal will allow us to manage this as reference configuration. @mikatammi - please have a look at the proposed changes to `flake.nix` and comment if you can think of other solutions.

Reply:
Pushed a new amendment with two more inputs two flake for nixpkgs-22.11 and nixos-hardware forked repository. Now other platforms will not be impacted. Please review.

Reply:
Proposed solution has conflicts with the HEAD of main 
![screencapture](https://github.com/tiiuae/ghaf/assets/1027150/9ba96fda-45fe-45bc-9187-eb38bc51f5bd)
and in addition it duplicates `nixpkgs` and `nixos-hardware` for all the targets. 
I'm afraid I can't support that solution. Please sync with @mikatammi and @brianmcgillion as they are rewriting the module system which has impact to reviewing / merging any other changes.